Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +659,157 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Track open
+
+    @MainActor
+    func test_onInlineMessageShown_expectTrackOpenedMetric_expectGlobalMessageShownListener() async {
+        let inlineView = await showInlineMessageForMetrics()
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Check if messageShown is called
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 1)
```

Comment:
In order to verify that a `opened` metric was tracked, we should verify that the eventbus was passed an event for tracking an `opened` metric. Not just check that some event was posted to it.